public/apps/account/account-app.tsx

```diff
@@ -71,7 +72,15 @@ export async function setupTopNavButton(coreStart: CoreStart, config: ClientConf
 
     let shouldShowTenantPopup = true;
 
-    if (tenantSpecifiedInUrl() || getShouldShowTenantPopup() === false) {
+    const dashboardsInfo = await getDashboardsInfoSafe(coreStart.http);
+    const multitenancyEnabled = dashboardsInfo?.multitenancy_enabled && config.multitenancy.enabled;
+    const defaultTenantSet = dashboardsInfo?.default_tenant !== '';
+    if (
+      tenantSpecifiedInUrl() ||
+      getShouldShowTenantPopup() === false ||
```

Comment:
I think we should rename some of these variables. `setShouldShowTenantPopup` looks like an optimization where the value to show or not show the tenant popup is computed once and stored in local storage and then always obtained from local storage. If this value is not correct, then it will always either show the tenant modal when its not supposed to or not show the tenant modal when its supposed to.

IMO The logic to compute whether the tenant modal should be displayed should be in a re-usable function:

1. If multi-tenancy is disabled then no need to display the modal.
2. If multi-tenancy is enabled then display the modal unless:
    - Tenant is specified in the URL
    - Default tenant is set

FYI `''` for the tenant means global. If there is any non-null value returned from the dashboardsInfo API then assume a default tenant is set.

Reply:
Not sure if I am understanding the setting correctly - when I did a fresh install of OpenSearch and the security plugin the default tenant is defaulted to '', so I think '' means no default tenant. Whereas if I set it to "Global", it is returning as "Global". However, it is confusing that the documentation states that global is defaulted - then do we never show it, since that is the default? What happens when global is not enabled?

Reply:
https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java#L603 - seems like this will give empty string instead of null in case it is not set